### PR TITLE
Refactored Testsuite setup

### PR DESCRIPTION
**Pull request by 0x20h**
Hi Klarnauts, 
I don't have phpunit installed :(
- Adds phpunit as a dev-requirement
- use standard phpunit.xml.dist (no need for `--configuration`)
- `__DIR__` is PHP >= 5.3
- use autoloader

To run tests now: 

```
php composer install
vendor/bin/phpunit
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/vendor
+/composer.lock
```
Comment (0x20h) [inline .gitignore]: A matter of taste to have it in the repo. I don't know, what do you think?
(https://getcomposer.org/doc/02-libraries.md#lock-file)

Comment (JoakimLofgren) [inline .gitignore]: When dealing with dependencies I prefer having the lock file committed as it simplifies troubleshooting.

**Comment by JoakimLofgren** [comment]
Thanks for your pull request. We will take a look at it.

**Comment by 0x20h** [comment]
Pretty neat now, thanks!

**Pull request merged**